server/opts.go

```diff
@@ -373,20 +373,54 @@ type token interface {
 // unwrapValue can be used to get the token and value from an item
 // to be able to report the line number in case of an incorrect
 // configuration.
-func unwrapValue(v interface{}) (token, interface{}) {
+// also stores the token in lastToken for use in convertPanicToError
+func unwrapValue(v interface{}, lastToken *token) (token, interface{}) {
 	switch tk := v.(type) {
 	case token:
+		if lastToken != nil {
+			*lastToken = tk
+		}
 		return tk, tk.Value()
 	default:
 		return nil, v
 	}
 }
 
+// use in defer to recover from panic and turn it into an error associated with last token
+func convertPanicToErrorList(lastToken *token, errors *[]error) {
+	// only recover if an error can be stored
+	if errors == nil {
+		return
+	} else if err := recover(); err == nil {
+		return
+	} else if lastToken != nil && *lastToken != nil {
+		*errors = append(*errors, &configErr{*lastToken, fmt.Sprint(err)})
+	} else {
+		*errors = append(*errors, fmt.Errorf("Encountered panic without a token %v", err))
+	}
+}
+
+// use in defer to recover from panic and turn it into an error associated with last token
+func convertPanicToError(lastToken *token, e *error) {
+	// only recover if an error can be stored
+	if e == nil || *e != nil {
```

Comment:
No action, but I got confused for a second here (because of the comment). Yes, we need `e` to be not nil so that we can store the constructed error (out of the panic) but we don't need to do anything if `*e` is not nil, which means that this function was invoked with an error (and likely not a panic).